src/modules/40-gitsync/pages/config/GitFullSyncEntityList.tsx

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright 2022 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import React from 'react'
+import type { CellProps, Column, Renderer } from 'react-table'
+import { Color, Icon, Layout, TableV2, Text } from '@harness/uicore'
+import type { IconProps } from '@harness/uicore/dist/icons/Icon'
+import { useStrings } from 'framework/strings'
+import type { PageGitFullSyncEntityInfoDTO, GitFullSyncEntityInfoDTO } from 'services/cd-ng'
+import { getEntityIconName } from '../entities/EntityHelper'
+import css from './GitSyncConfigTab.module.scss'
+
+interface GitFullSyncEntityListProps {
+  data: PageGitFullSyncEntityInfoDTO
+  gotoPage: (pageNumber: number) => void
+}
+
+interface StatusData {
+  icontProp: IconProps
+  statusBackground?: Color
+  highlightedColor?: Color
+}
+
+const RenderEntityStatus = (prop: Record<'status', GitFullSyncEntityInfoDTO['syncStatus']>): JSX.Element => {
+  const status = prop.status
+  let data: StatusData
+
+  switch (prop.status) {
```

Comment:
Might as well use the `status` variable from a few lines above
```suggestion
  const status = prop.status
  let data: StatusData

  switch (status) {
```